lib/llvmopencl/WorkitemLoops.cc

```diff
@@ -979,14 +1089,70 @@ void WorkitemLoopsImpl::addContextSaveRestore(llvm::Instruction *Def) {
       assert(IncomingBB != NULL);
       ContextRestoreLocation = IncomingBB->getTerminator();
     }
-    llvm::Value *LoadedValue =
-        addContextRestore(UserI, Alloca, Def->getType(), PaddingAdded,
-                          ContextRestoreLocation, isa<AllocaInst>(Def));
-    UserI->replaceUsesOfWith(Def, LoadedValue);
+
+    llvm::Value *LocalValue = nullptr;
```

Comment:
Would it be more clear if this variable were called `RematerializedValue`?

Reply:
Originally the idea was to use it for both remat and normal context restored value, but yep, I'll refactor it to make this spot more readable.